bundles/org.eclipse.osgi.tests/src/org/eclipse/osgi/tests/services/datalocation/BasicLocationTests.java

```diff
@@ -142,19 +159,21 @@ public void testCreateLocation03() {
 		testLocation.release();
 	}
 
+	@Test
 	public void testCreateLocation04() {
 		Location configLocation = configLocationTracker.getService();
 		File testLocationFile = OSGiTestsActivator.getContext().getDataFile("testLocations/testCreateLocation04");
 		Location testLocation = configLocation.createLocation(null, null, true);
 		try {
 			testLocation.set(testLocationFile.toURL(), true);
 			testLocation.release();
-			fail("Should not be able to lock read-only location");
+			Assert.fail("Should not be able to lock read-only location");
```

Comment:
Could not be replaced with an `assertThrows` as this test actually fails silently, see #460